lib/SSH/LibSSH.pm6

```diff
@@ -350,13 +351,50 @@ class SSH::LibSSH {
             }
         }
 
-        method !process-auth-outcome($outcome, $v) {
+        method !process-auth-outcome($outcome, $v, :$method = "key") {
             if $outcome == SSH_AUTH_SUCCESS {
                 $v.keep(self);
             }
             else {
-                self!teardown-session();
-                $v.break(X::SSH::LibSSH::Error.new(message => 'Authentication failed'));
+                if $method eq "key" and defined $.password {
```

Comment:
I've tended to use `$!foo` elsewhere in the class rather than the accessor form, could be good to make it `$!password` here and elsewhere to match. I also worry a bit about *having* a `.password` accessor; it's just asking for it to get leaked into some `.perl` output when debugging or logging or something, so maybe we should just keep it private?